### ISSUE TITLE
`$` in pathname gets transformed and the file cannot be found anymore

### ISSUE THREAD

**Issue by ivanhofer** [prettier/prettier#13258]
**Environments:**

- Prettier Version: 2.7.1
- Running Prettier via: CLI
- Runtime: Node.js v16.15.1
- Operating System: Ubuntu 20.04 (Windows 10 with WSL 2)
- Prettier plugins (if any): none

**Steps to reproduce:**

Run following command
`npx prettier "/home/user/project/src/$state/builder-form.ts"`

**Expected behavior:**

`prettier` should not transform anything between the quotes (`"`)

**Actual behavior:**

`prettier` removes everything from the `$` until the next slash `/` and removes it from the path. I get following error:
```
[error] No files matching the pattern were found: "/home/user/project/src//builder-form.ts".
```

**Comment by thorn0** [comment]
Does this file exist?

**Comment by JounQin** [comment]
<img width="741" alt="image" src="https://user-images.githubusercontent.com/8336744/183469679-dfe60a9a-a97d-4d40-91eb-da5228346ea2.png">

This is how node work.

**Comment by ivanhofer** [comment]
Also by escaping the `$` sign I get the same error.
`npx prettier "/home/user/project/src/\$state/builder-form.ts"`

`[error] No files matching the pattern were found: "/home/user/project/src//builder-form.ts".`

**Comment by JounQin** [comment]
> Also by escaping the `$` sign I get the same error. `npx prettier "/home/user/project/src/\$state/builder-form.ts"`
> 
> `[error] No files matching the pattern were found: "/home/user/project/src//builder-form.ts".`

I do confirm this by a simple `$test.js` file and `npx prettier --write "\$test.js"`.

**Comment by JounQin** [comment]
<img width="701" alt="image" src="https://user-images.githubusercontent.com/8336744/183474164-03253257-25a8-4f19-a77b-2cafe6e9626f.png">

It should be a npx bug.

**Comment by JounQin** [comment]
I just raised https://github.com/nodejs/node/issues/44178 to track.

I think it can be closed here. @ivanhofer

**Comment by ivanhofer** [comment]
@JounQin I can confirm.
Thanks for your help and investigation!

**Comment by thorn0** [comment]
It's how the shell works.
```sh
$ echo "Hello $name!"
Hello !
$ echo 'Hello $name!'
Hello $name
```
Use single quotes to prevent interpolation.

**Comment by JounQin** [comment]
@thorn0 You didn't look through the whole issue. 😅

**Comment by thorn0** [comment]
I did and couldn't reproduce anything wrong, at least on Ubuntu. Macos's shell might work slightly differently.

**Comment by JounQin** [comment]
> I did and couldn't reproduce anything wrong, at least on Ubuntu. Macos's shell might work slightly differently.

Maybe that's because it has been fixed on `npm@8.15.0`

I can confirm when using `npm@8.16.0`, it works as expected now.
